lib/src/screens/card-details/card_options_screen.dart

```diff
@@ -0,0 +1,214 @@
+import 'dart:io';
+import 'package:flutter/material.dart';
+import 'package:bugsnag_flutter/bugsnag_flutter.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:antdesign_icons/antdesign_icons.dart';
+import 'package:cartan/utils/app_snackbar.dart';
+import 'package:provider/provider.dart';
+import 'package:url_launcher/url_launcher.dart';
+import 'package:cartan/src/models/merchant.dart';
+import 'package:cartan/src/models/loyalty_card.dart';
+import 'package:cartan/src/repositories/loyalty_card_repository.dart';
+
+class CardOptionsScreen extends StatelessWidget {
+  final LoyaltyCard card;
+  final Merchant merchant;
+
+  const CardOptionsScreen({
+    super.key,
+    required this.card,
+    required this.merchant,
+  });
+
+  void _showDeleteDialog(BuildContext context, String cardId) {
+    final localizations = AppLocalizations.of(context)!;
+
+    showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: Text(localizations.confirm_delete),
+          content: Text(localizations.confirm_delete_message),
+          actions: <Widget>[
+            TextButton(
+              child: Text(localizations.cancel),
+              onPressed: () {
+                Navigator.of(context).pop();
+              },
+            ),
+            TextButton(
+              style: TextButton.styleFrom(
+                foregroundColor: Theme.of(context).colorScheme.error,
+              ),
+              onPressed: () async {
+                try {
+                  final cardRepo = Provider.of<LoyaltyCardRepository>(context, listen: false);
+                  await cardRepo.deleteCard(cardId);
```

Comment:
```suggestion
                  Provider.of<LoyaltyCardRepository>(context, listen: false).deleteCard(cardId);
```

não há necessidade de separar em variavel

Reply:
tens toda a razão, alterado